src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,60 @@
+@use "../utils/_variables" as *;
+@use "../utils/_functions" ;
+@use "../utils/_mixins" ;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $cell-gap;
+  padding: $calendar-padding;
+  max-width: 7 * $cell-size + 2 * $calendar-padding + 6 * $cell-gap;
+  width: 100%;
+
+  @each $day-count, $index in $week {
+      &--start-day_#{$day-count} {
+      .calendar__day:first-of-type {
+        margin-left: functions.getShift($cell-size, $cell-gap, $index);
+      }
+    }
+  }
+
+  @for $day from 28 through 31 {
+    &--month-length_#{$day} {
+      @for $i from $day through 31 {
+        .calendar__day:nth-child(#{$i + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &__day {
+    position: relative;
+    @include mixins.square($cell-size);
+    background-color: #eee;
+    border: 1px solid black;
+    transition: .5s ease-in-out;
+    display: inline-block;
+
+    &:hover {
+      transform: translateY(-($cell-size * 0.2));
+      background-color: #ffbfcb;
+      transition: .5s ease-in-out;
+      cursor: pointer;
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}):before {
+        content: "#{$i}";
+        position: relative;
+        top: 50%;
+        transform: translateY(-50%);
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        font-size: 30px;
```

Comment:
<img width="797" alt="image" src="https://user-images.githubusercontent.com/102411130/230849286-52858ec8-79f9-47c1-950b-21129445de7d.png">
no need to repeat all styles in loops

only `content` is being changed here